### PR TITLE
Support iframe embed in readable version.

### PR DESCRIPTION
**Pull request by snow01**
1) Support https URL for iframe embeds.
2) Support external setting for allowed URLs.
3) Emit iframe embed in readable version.
4) Fix getURL path in bin/cli.js
5) Added .gitignore

### CODE REVIEW

File: bin/cli.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ if(process.argv.length < 3 || !/^https?:\/\//.test(process.argv[2])){
 	return;
 }
 
-require("./getURL.js")(process.argv[2], process.argv[3] === "html" ? "html" : "text", function(result){
+require("./../lib/getURL.js")(process.argv[2], process.argv[3] === "html" ? "html" : "text", function(result){
```
Comment (fb55) [inline bin/cli.js]: `"../lib/…`

File: readabilitySAX.js [hunk 2]
```diff
@@ -150,7 +150,7 @@ var tagsToSkip = {__proto__:null,aside:true,footer:true,head:true,label:true,nav
     divToPElements = ["a","blockquote","dl","img","ol","p","pre","table","ul"],
     okayIfEmpty = ["audio","embed","iframe","img","object","video"],
 
-    re_videos = /http:\/\/(?:www\.)?(?:youtube|vimeo)\.com/,
+    re_videos = /https?:\/\/(?:www\.)?(?:youtube|vimeo)\.(?:com)/,
```
Comment (fb55) [inline readabilitySAX.js]: The anonymous group at the end isn't needed.

File: readabilitySAX.js [hunk 3]
```diff
@@ -206,7 +206,10 @@ Readability.prototype._settings = {
 	linksToSkip: {},	//pages that are already parsed
 	//pageURL: null,	//URL of the page which is parsed
 	//type: "html",		//default type of output
-	resolvePaths: false
+	resolvePaths: false,
+    includeEmbeds: function(src) {
```
Comment (fb55) [inline readabilitySAX.js]: This shouldn't replace the `embed` handling, but might be added as a separate option.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,5 @@
+.DS_Store
```
Comment (fb55) [inline .gitignore]: Please don't add a `.gitignore`.

File: bin/cli.js (hunk 1 above)
Comment (fb55) [inline bin/cli.js]: Should also be a separate commit.

File: readabilitySAX.js [hunk 5]
```diff
@@ -491,18 +494,18 @@ Readability.prototype.onclosetag = function(tagName){
 	else if(this._settings.cleanConditionally && tagName in cleanConditionally){
 		var p = elem.info.tagCount.p || 0,
 		    contentLength = elem.info.textLength + elem.info.linkLength;
-
+
```
Comment (fb55) [inline readabilitySAX.js]: Please don't add random whitespace.

File: readabilitySAX.js [hunk 6]
```diff
@@ -491,18 +494,18 @@ Readability.prototype.onclosetag = function(tagName){
...
 		    contentLength = elem.info.textLength + elem.info.linkLength;
-
+        
 		if(contentLength === 0){
 			if(elem.children.length === 0) return;
 			if(elem.children.length === 1 && typeof elem.children[0] === "string") return;
 		}
 		if((elem.info.tagCount.li - 100) > p && tagName !== "ul" && tagName !== "ol") return;
-		if(contentLength < 25 && (!("img" in elem.info.tagCount) || elem.info.tagCount.img > 2) ) return;
+		if(contentLength < 25 && ((!("img" in elem.info.tagCount) && !("iframe" in elem.info.tagCount)) || elem.info.tagCount.img > 2 || elem.info.tagCount.iframe > 2) ) return;
```
Comment (fb55) [inline readabilitySAX.js]: Not sure I like this, the image part is often annoying on its own.

It might be interesting to see how Safari Reader is handling this. I wanted to have a look at their algorithm changes for quite some time now, this might be an interesting starting point.

**Comment by fb55** [comment]
Thanks for the pull request! I left some comments, it would be great if you could have a look.

File: readabilitySAX.js (hunk 3 above)
Comment (snow01) [inline readabilitySAX.js]: Default method implementation does not override the existing embed handling. This setting is provided to support, if user wants to allow a different set of embeds than supported by re_videos regular expression. Or did you mean something else ?

File: readabilitySAX.js (hunk 2 above)
Comment (snow01) [inline readabilitySAX.js]: Yep. Left their mistakenly, added that for slideshare.net - but later removed that site from regular expression.

File: bin/cli.js (hunk 1 above)
Comment (snow01) [inline bin/cli.js]: Is it okay for it to be in this commit, or you still want it to be a separate commit ?

File: .gitignore (hunk 4 above)
Comment (snow01) [inline .gitignore]: Okay. Most projects on github add .gitignore - if you still want it to be removed, will remove it.

File: readabilitySAX.js (hunk 5 above)
Comment (snow01) [inline readabilitySAX.js]: Okay.

File: readabilitySAX.js (hunk 6 above)
Comment (snow01) [inline readabilitySAX.js]: I tested on 10-15 websites and embeds are working correct.

Reg: Safari Reader, or Firefox Reader, they do not include embeds on many readable page. But in my opinion default embed put in the normal flow of article shall be included. If they are not doing today, they will do tomorrow.

Take this page as an example -- http://www.inbound.com/blog/charlene-li-ideas-and-experiences

Embeds are in the normal flow of the article - in readable version they shall be shown, in my opinion.

Comment (snow01) [inline readabilitySAX.js]: Reg: image selection part of the algorithm -- I know it was very hard to understand the logic. It could be simplified, but most heuristics implementation would look similar ugly.

Only questionable aspect was why 25 content length. Probably we can modify algorithm to 2 pass --

1) In pass 1 - collect average statistics for the page. For example, average length of text for image on that particular page.
2) In pass 2 - use these statistics to score images.

Probably such statistics based calculation may cover wider cases. And if we want to really make it comprehensive, probably for some users, statistics can be stored in some key-value DB - across websites, for the website.

Comment (snow01) [inline readabilitySAX.js]: Reg image selection -- your algorithm works for most cases. But one specific case is being left -- when main image of the page is before title. I see there are 3 possibilities for the main image --

1) Before the title.
2) Just after title and before the other content of page.
3) In the body.

I see for many websites "before the title" image is not getting included.

**Comment by fb55** [comment]
Comments not addressed, closing
